src/cli/commands/force/lightning/lwc/preview.ts

```diff
@@ -211,80 +267,69 @@ export default class Preview extends Setup {
         return Promise.resolve();
     }
 
-    public launchPreview(): Promise<boolean> {
-        const platform = this.flags.platform;
-
-        const defaultDeviceName = CommandLineUtils.platformFlagIsIOS(platform)
-            ? iOSConfig.defaultSimulatorName
-            : androidConfig.defaultEmulatorName;
-
-        const device = CommandLineUtils.resolveFlag(
-            this.flags.target,
-            defaultDeviceName
-        );
-
-        const targetApp = CommandLineUtils.resolveFlag(
-            this.flags.targetapp,
-            PreviewUtils.BROWSER_TARGET_APP
-        );
-
-        const projectDir = CommandLineUtils.resolveFlag(
-            this.flags.projectdir,
-            process.cwd()
-        );
+    public async isLwcServerRunning(): Promise<string> {
+        const port = CommonUtils.getLwcServerPort();
+        if (!port) {
+            return Promise.reject(this.unfulfilledMessage);
+        } else {
+            return Promise.resolve(util.format(this.fulfilledMessage, port));
+        }
+    }
 
-        const configFileName = CommandLineUtils.resolveFlag(
-            this.flags.configfile,
-            ''
-        );
+    public launchPreview(): Promise<boolean> {
+        // At this point all of the inputs/parameters have been verified and parsed so we can just use them.
```

Comment:
In `validateAdditionalInputs` we already parse the input params to verify them. So I decided to cache the results of `validateAdditionalInputs` so that I wouldn't need to re-parse here again and this made the code here much simpler.